rfc/src/rfcs/0008-source-coverage.md

```diff
@@ -0,0 +1,300 @@
+It's not rare to see the term "region coverage" being used instead to refer to the same thing.
+That's because LLVM's source-based code coverage feature can report coverage for code regions,
+but other coverage frameworks don't support the concept of code regions.
+
+### The Kani coverage workflow
+
+The proposed Kani coverage workflow imitates the LLVM coverage workflow as much as possible.
+
+The two main components of the Kani coverage workflow will be the following:
+ 1. A new subcommand `cov` that drives the coverage workflow in Kani and
+    produces machine-readable coverage results.
+ 2. A new tool `kani-cov` that consumes the machine-readable coverage results
+    emitted by Kani to produce human-readable results in the desired format(s).
+
+Therefore, the first part of the coverage workflow will be managed by Kani.
+Then, in the other part, we will use the `kani-cov` tool to produce the coverage
+output(s) we're interested in.
+
+In the following, we describe each one of these components in more detail.
+
+#### The `-cov` option
+
+The coverage workflow will be kicked off through a new `-cov` option:
+
+```sh
+cargo kani -cov
+```
+
+The main difference with respect to the regular verification workflow is that,
+at the end of the verification-based coverage run, Kani will generate two types
+of files:
+ - One single file `.kanimap` file for the project. This file will contain the
+ coverage mappings for the project's source code.
+ - One `.kaniraw` file for each harness. This file will contain the
+ verification-based results for the coverage-oriented properties corresponding
+ to a given harness.
+
+Note that `.kaniraw` files correspond to `.profraw` files in the LLVM coverage
+workflow. Similarly, the `.kanimap` file corresponds to the coverage-related
+information that's embedded into the project's binaries in the LLVM coverage
+workflow.[^note-kanimap]
+
+The files will be written into a new timestamped directory associated with the
+coverage run. The path to this directory will be printed to standard output in
+by default. For example, the [draft implementation](https://github.com/model-checking/kani/pull/3119)
+writes the coverage files into the `target/kani/<target_triple>/cov/` directory.
+
+Users aren't expected to read the information in any of these files.
+Therefore, there's no need to restrict their format.
+The [draft implementation](https://github.com/model-checking/kani/pull/3119)
+uses the JSON format but we might consider using a binary format if it doesn't
+scale.
+
+[^note-kanimap]: Note that the `.kanimap` generation isn't implemented in [#3119](https://github.com/model-checking/kani/pull/3119).
+The [draft implementation of `kani-cov`](https://github.com/model-checking/kani/pull/3121)
+simply reads the source files referred to by the code coverage checks, but it
+doesn't get information about code trimmed out by the MIR linker.
+
+#### The `kani-cov` tool
```

Comment:
I am curious on why this is a separate command instead of a sub-command.

Reply:
Yes, it'd make a lot of sense to have a sub-command instead. Calling this from `kani-driver` as you mention in the other thread is likely the best choice, but it's not clear to me if should have a binary or just a library for `kani-cov`. What do you think?